network/strategy.go

```diff
@@ -26,9 +28,21 @@ type NetworkStrategy interface {
 // provided type.  If no strategy is registered for the type an
 // ErrNotValidStrategyType is returned.
 func GetStrategy(tpe string) (NetworkStrategy, error) {
+	strategiesMtx.RLock()
+	defer strategiesMtx.RUnlock()
 	s, exists := strategies[tpe]
 	if !exists {
 		return nil, ErrNotValidStrategyType
 	}
 	return s, nil
 }
+
+// AddStrategy registers a network strategy to be used for the
+// provided type. If there is a strategy already associated with
+// that type, it will be overridden. Multiple goroutines can
+// safely call AddStrategy.
+func AddStrategy(tpe string, strategy NetworkStrategy) {
+	strategiesMtx.Lock()
+	defer strategiesMtx.Unlock()
```

Comment:
I don't see why you'd need to use a `defer` here. This might just be nitpicking, but it looks weird to have a `defer` for an operation such as this.